src/main/scala/com/snowflake/snowpark/internal/analyzer/SnowflakePlanNode.scala

```diff
@@ -18,6 +19,15 @@ private[snowpark] trait LogicalPlan {
     (analyzedPlan, analyzer.getAliasMap)
   }
 
+  var dfAliasMap: MMap[String, Seq[Attribute]] = MMap.empty
```

Comment:
`LogicalPlan` should be immutable. So don't change it after creation, move this to the constructor like I mentioned above